### PR TITLE
optimize `extract_repo()` and add test

### LINKED ISSUE #114: extract_repo is unable to handle 'www' in repository urls
Hi All, I was trying to use {travis} and came across an this issue. I looked at the underlying function, and it looked like there was some functionality to escape the 'www', but appeared not to be working. I also did not see the tests using 'www' in their URLs. 

Below is a reprex:
``` r
travis:::extract_repo("https://www.github.com/thebioengineer/tidytuesdayR.git")
#> Error: Unrecognized repo format: https://www.github.com/thebioengineer/tidytuesdayR.git
```

<sup>Created on 2019-11-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by pat-s**
fixes #114

### CODE REVIEW

File: R/remote.R [hunk 1]
```diff
@@ -13,18 +13,41 @@ get_remote_url <- function(path) {
...
+
+  # account for ssh notation
   if (grepl("^git@github.com:", url)) {
     url <- sub("^git@github.com:", "https://github.com/", url)
+  } else if (grepl("^git://github.com:", url)) {
+    url <- sub("^git://github.com:", "https://github.com/", url)
   } else if (grepl("^git://github.com", url)) {
-    url <- sub("^git://github.com", "https://github.com", url)
-  } else if (grepl("^http://(.+@)?github.com", url)) {
+    url <- sub("^git://github.com", "https://github.com/", url)
```
Comment (krlmlr) [inline R/remote.R]: This looks odd.

Comment (pat-s) [reply R/remote.R]: Can you add more context to this?

**Comment by pat-s** [comment]
> Can we remove "^.*[.:/]github[.]com[:/]" and use the rest (minus the optional .git suffix) as repo identifier? We can fail if the attempt to remove didn't change anything.

I simplified `extract_repo()` a lot. 
Not sure if its full fool-proof but the tests like it ;)

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**